### PR TITLE
buildbot/master.cfg: Implement macOS Updater support

### PR DESCRIPTION
**Pull request by spycrab**
Update ``master.cfg`` to generate manifests for macOS.

### CODE REVIEW

File: buildbot/master.cfg [hunk 1]
```diff
@@ -340,10 +341,16 @@ def make_dolphin_osx_build(mode="normal"):
...
+                           workdir="build/build",
+                           description="signing updater",
+                           descriptionDone="sign updater",
+                           haltOnFailure=True))
 
     f.addStep(ShellCommand(command=["hdiutil", "create", "dolphin.dmg",
                                     "-format", "UDBZ",
-                                    "-srcfolder", "Binaries/dolphin.app", "-ov",
+                                    "-layout", "SPUD", # Needed for 7-Zip support
+                                    "-srcfolder", "Binaries/Dolphin.app", "-ov",
```
Comment (shuffle2) [inline buildbot/master.cfg]: does case of "dolphin" matter here?

Comment (spycrab) [reply buildbot/master.cfg]: I do think so. It's probably the cause for ``dolphin.app`` not being capitalized as ``Dolphin.app`` as it should.

Comment (shuffle2) [reply buildbot/master.cfg]: I meant: "-srcfolder Binaries/Dolphin.app" implies "Binaries/Dolphin.app" already exists. Is that the case? Or this change runs risk of breaking things.

Comment (spycrab) [reply buildbot/master.cfg]: Used before with codesign.sh. pretty sure it won't cause issues

File: buildbot/master.cfg [hunk 2]
```diff
@@ -387,6 +396,24 @@ def make_dolphin_osx_build(mode="normal"):
                                      description="notifying website",
                                      descriptionDone="website notice"))
 
+        tmp_filename = "/tmp/%s.7" % (os.path.basename(master_filename))
```
Comment (delroth) [inline buildbot/master.cfg]: 7z

File: buildbot/master.cfg (hunk 1 above)
Comment (spycrab) [reply buildbot/master.cfg]: Might be caused by us using APFS which is not case sensitive...

**Pull request merged**